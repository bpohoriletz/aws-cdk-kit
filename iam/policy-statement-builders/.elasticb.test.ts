import ElasticBPolicyStatementBuilder from './elasticb';

describe('new ElasticBPolicyStatementBuilder', () => {
  let instance: ElasticBPolicyStatementBuilder;

  beforeEach(() => {
    instance = new ElasticBPolicyStatementBuilder();
  });

  test('has correct defaults set', () => {
    expect(instance).toMatchSnapshot();
  });

  test('.setUploadEC2Stats allows sending EC2 stats', () => {
    instance.setUploadEC2Stats();

    expect(instance).toMatchSnapshot();
  });

  test('.addResource adds resource', () => {
    instance.addResource('*');

    expect(instance).toMatchSnapshot();
  });
});
