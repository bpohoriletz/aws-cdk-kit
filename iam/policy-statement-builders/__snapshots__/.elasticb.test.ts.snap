// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new ElasticBPolicyStatementBuilder .addResource adds resource 1`] = `
ElasticBPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [
      "*",
    ],
  },
}
`;

exports[`new ElasticBPolicyStatementBuilder .setPermissions allows sending EC2 stats 1`] = `
ElasticBPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [
      "elasticbeanstalk:PutInstanceStatistics",
    ],
    "resources": [],
  },
}
`;

exports[`new ElasticBPolicyStatementBuilder has correct defaults set 1`] = `
ElasticBPolicyStatementBuilder {
  "policyStatementProps": PolicyStatementProduct {
    "actions": [],
    "resources": [],
  },
}
`;
